overlord/ifacestate/ifacestate_test.go

```diff
@@ -5568,11 +5600,106 @@ func (s *interfaceManagerSuite) TestGadgetConnect(c *C) {
 	c.Assert(gotConnect, Equals, true)
 }
 
-func (s *interfaceManagerSuite) TestGadgetConnectAlreadyConnected(c *C) {
+func (s *interfaceManagerSuite) TestAutoConnectGadgetProducer(c *C) {
 	r1 := release.MockOnClassic(false)
 	defer r1()
 
-	s.setupGadgetConnect(c)
+	s.setupAutoConnectGadget(c)
+	s.manager(c)
+
+	s.state.Lock()
+	defer s.state.Unlock()
+
+	chg := s.state.NewChange("setting-up", "...")
+	t := s.state.NewTask("auto-connect", "gadget connections")
+	t.Set("snap-setup", &snapstate.SnapSetup{
+		SideInfo: &snap.SideInfo{
+			RealName: "producer"},
+	})
+	chg.AddTask(t)
+
+	s.state.Unlock()
+	s.se.Ensure()
+	s.se.Wait()
+	s.state.Lock()
+
+	c.Assert(chg.Err(), IsNil)
+	tasks := chg.Tasks()
+	c.Assert(tasks, HasLen, 7)
```

Comment:
Can we have checks for all the 7 tasks expected here?

Reply:
done by extracting a helper from the previous test

Reply:
Thank you!